Sources/MongoSwift/BSON/Document+Sequence.swift

```diff
@@ -242,6 +247,18 @@ public class DocumentIterator: IteratorProtocol {
         return values
     }
 
+    /// Returns the current value (equivalent to the `currentValue` property) or throws on error.
+    internal func safeCurrentValue() throws -> BSONValue? {
+        switch self.currentType {
+        case .symbol:
+            return try Symbol.asString(from: self)
+        case .dbPointer:
+            return try DBPointer.asDocument(from: self)
+        default:
+            return try DocumentIterator.bsonTypeMap[currentType]?.init(from: self)
```

Comment:
Looking at DBPointer and Symbol, I understand they're handled explicitly above. Their `init(from iter: DocumentIterator)` method always throws because we don't want users creating these types anew. Is there a reason we don't handle the `.undefined` case anywhere other than `Document.setValue()` when we check for a no-op replacement?

It's also conspicuously absent from `DocumentIterator.bsonTypeMap`. If `currentType` doesn't exist in `bsonTypeMap` here, does the `return` statement simply evaluate the `nil`? Perhaps this is covered under "Accessing Subscripts of Optional Type" in https://docs.swift.org/swift-book/LanguageGuide/OptionalChaining.html.

Reply:
I *think* the implicit behavior here is that undefined is just upconverted to a nil

Reply:
Yeah, Kaitlin should be write, this should return `nil` for `.undefined`.